src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
+        {
+            // If it is a document selection, we'll retrieve the text from the document
+            ExecuteDocumentSelectionParams docRequest = request as ExecuteDocumentSelectionParams;
+            if (docRequest != null)
+            {
+                // Get the document from the parameters
+                ExecuteDocumentSelectionParams documentRequest = (ExecuteDocumentSelectionParams) request;
```

Comment:
Also to be nitpicky - this is a bunch of independent logic in an If clause. For clarity, why not move everything in the if clause into its own method and just call `return GetSqlTextFromDocRequest`? That'll make it easier to read this method since we can see the overall logic there and ignore this detailed lookup logic.